classes/collector/manager.php

```diff
@@ -49,9 +49,15 @@ public static function send_metrics(array $items) {
             try {
                 if ($collector->is_ready()) {
                     $collector->record_metrics($items);
+                    // Clear any failure timestamps for this plugin.
+                    unset_config($plugin->name, 'tool_cloudmetrics_failure');
```

Comment:
Maybe it is better to store an explicit 'pass' so we can detect the different between 'unknown' vs 'good' in the check we expose

Also check if we need to set it and only set it if needed